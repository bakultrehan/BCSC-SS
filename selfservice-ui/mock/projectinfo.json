{
  "project-info": [
    {
      "id": 1,
      "organizationName": "organic",
      "projectName": "project",
      "myRole": "1",
      "developerDetails": {
        "email": "test@test.com",
        "phone": "0987654321",
        "firstName": "name",
        "lastName": "l name",
        "id": 1
      },
      "managerDetails": {
        "email": "mail@test.com",
        "phone": "1234567890",
        "firstName": "f manager",
        "lastName": "l manager",
        "id": 2
      },
      "ctoDetails": {
        "email": "cto@email.com",
        "phone": "1234567890",
        "firstName": "f cto",
        "lastName": "l cto",
        "id": 3
      }
    },
    {
      "organizationName": "aa",
      "projectName": "aa",
      "description": "testaa",
      "myRole": "1",
      "developerDetails": {},
      "managerDetails": {
        "firstName": "mf",
        "lastName": "ml",
        "email": "me",
        "phone": "mp"
      },
      "ctoDetails": {
        "firstName": "cf",
        "lastName": "cl",
        "email": "ce",
        "phone": "cp"
      },
      "id": 11
    },
    {
      "organizationName": "ss1",
      "projectName": "dd",
      "description": "test",
      "myRole": "1",
      "developerDetails": {},
      "managerDetails": {
        "firstName": "mf",
        "lastName": "m",
        "email": "m",
        "phone": "m"
      },
      "ctoDetails": {
        "firstName": "c",
        "lastName": "c",
        "email": "c",
        "phone": "c"
      },
      "id": 12
    }
  ]
}