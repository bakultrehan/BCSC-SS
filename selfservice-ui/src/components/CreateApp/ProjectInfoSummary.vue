/** * ProjectInfoSummary component */

<template>
  <v-card>
    <Loading v-if="isLoading" />
    <v-toolbar dense class="bc-subtitle-2" dark v-else>
      <v-card-title>
        {{ $t('summaryPage.projectInfoTitle') }}
        <v-spacer></v-spacer>
        <span
          @click="$router.push(`/project/${id}/info`)"
          class="edit-wrapper"
          :aria-label="$t('global.edit')"
        >
          <v-icon small class="ml-3">mdi-pencil</v-icon>
          <span class="edit-label">{{ $t('global.edit') }}</span>
        </span>
      </v-card-title>
    </v-toolbar>

    <v-list dense class="px-5" v-if="!isLoading">
      <v-list-item>
        <v-list-item-content class="pr-30">
          {{ $t('summaryPage.labelOrganizationName') }}
          <span
            class="small-hint pad-50"
            v-html="$t('summaryPage.OrganizationNameHint')"
          ></span>
        </v-list-item-content>
        <v-list-item-content class="align-end">
          <div>
            <v-icon small class="mr-1">mdi-office-building</v-icon>
            {{ projectInfo && projectInfo.organizationName }}
          </div>
        </v-list-item-content>
      </v-list-item>
      <v-divider></v-divider>
      <v-list-item>
        <v-list-item-content class="pr-30">
          {{ $t('summaryPage.labelProjectTitle') }}
          <span
            class="small-hint pad-50"
            v-html="$t('summaryPage.projectNameHint')"
          ></span>
        </v-list-item-content>
        <v-list-item-content class="align-end">
          <div>
            <v-icon small class="mr-1">mdi-apps</v-icon>
            {{ projectInfo.projectName }}
          </div>
        </v-list-item-content>
      </v-list-item>
      <v-divider></v-divider>
      <v-list-item>
        <v-list-item-content class="align-self-start pr-30">
          {{ $t('summaryPage.labelProjectDescription') }}
          <span
            class="small-hint pad-50"
            v-html="$t('summaryPage.labelProjectDescriptionHint')"
          ></span>
        </v-list-item-content>
        <v-list-item-content class="align-end">
          <div class="break-word-all">
            <v-icon small class="mr-1">mdi-apps</v-icon>
            {{ projectInfo.description }}
          </div>
        </v-list-item-content>
      </v-list-item>
    </v-list>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Loading from '@/Atomic/Loading/Loading.vue';

const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({
  components: { Loading }
})
export default class ProjectInfoSummary extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public projectInfo!: any;
  @ProjectInfoModule.Action('loadSingleProjectInfo')
  public loadSingleProjectInfo!: any;

  @ProjectInfoModule.Getter('getFinalProjectSubmissionStatus')
  public getFinalProjectSubmissionStatus!: any;

  @ProjectInfoModule.Action('clearSubmitProjectStatus')
  public clearSubmitProjectStatus!: any;

  private isLoading: boolean = true;

  @Watch('projectInfo')
  private ongetprojectInfoChanged(val: any) {
    this.isLoading = false;
  }

  private mounted() {
    this.isLoading = true;
    this.loadSingleProjectInfo(this.id);
  }
}
</script>
<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
</style>
