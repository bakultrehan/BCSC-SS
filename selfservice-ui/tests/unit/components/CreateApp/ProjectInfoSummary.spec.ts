import { shallowMount } from '@vue/test-utils';
import ProjectInfoSummary from '@/components/CreateApp/ProjectInfoSummary.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

import { singleProject } from '../../mocks/projectinfo';

let vuetify: any;
vuetify = new Vuetify();

describe('ProjectInfoSummary.vue', () => {
  const storeData = new Vuex.Store({
    modules: {
      ProjectInfoModule: {
        namespaced: true,
        state: {},
        getters: {
          getSingleProjectInfo: jest.fn(() => {
            return {
              singleProject
            };
          }),
          getFinalProjectSubmissionStatus: jest.fn(),
          errorStatus: jest.fn()
        },
        actions: {
          loadSingleProjectInfo: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return shallowMount(ProjectInfoSummary, {
      store: storeData,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed with gettors', () => {
    const projectInfoSummary = mountFunction({});

    expect(projectInfoSummary.element).toMatchSnapshot();
  });
});
