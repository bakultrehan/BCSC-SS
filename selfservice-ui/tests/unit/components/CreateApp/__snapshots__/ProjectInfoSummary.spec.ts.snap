// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfoSummary.vue renders props when passed with gettors 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="container"
  >
    <div
      class="layout d-flex align-center justify-center row wrap"
    >
      <div
        class="mt-10"
        col="4"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="v-progress-linear loading v-progress-linear--rounded theme--light"
          role="progressbar"
          style="height: 6px;"
        >
          <div
            class="v-progress-linear__background primary"
            style="opacity: 0.3; left: 0%; width: 100%;"
          />
          <div
            class="v-progress-linear__buffer"
          />
          <div
            class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active"
          >
            <div
              class="v-progress-linear__indeterminate long primary"
            />
            <div
              class="v-progress-linear__indeterminate short primary"
            />
          </div>
        </div>
         
        <span />
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;
